### PR TITLE
[API][Web][SIMS-#696]-MInistry Edit Location - Institution Location controller refactor

### PR DESCRIPTION
**Pull request by guru-aot**
Refactoring institution-location controller, due to route conflicts on locationId and institutionId get call.
Removed disabling inputs for institution user Add location.

### CODE REVIEW

File: sources/packages/api/src/route-controllers/institution-locations/institution-location.aest.controller.ts [hunk 1]
```diff
@@ -28,28 +27,13 @@ export class InstitutionLocationAESTController extends BaseController {
...
-  ): Promise<InstitutionLocationAPIOutDTO[]> {
-    // get all institution locations with designation statuses.
-    return this.locationControllerService.getInstitutionLocations(
-      institutionId,
-    );
-  }
-
   /**
    * Controller method to retrieve institution location by id.
    * TODO: updating of API routes will be handled in PART 2 PR
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution-locations/institution-location.aest.controller.ts:32]: is this part 2? 😉

File: sources/packages/api/src/route-controllers/institution-locations/institution-location.aest.controller.ts [hunk 2]
```diff
@@ -18,7 +17,7 @@ import {
  */
 @AllowAuthorizedParty(AuthorizedParties.aest)
 @Groups(UserGroups.AESTUser)
-@Controller("institution/location")
+@Controller("location")
```
Comment (ann-aot) [inline sources/packages/api/src/route-controllers/institution-locations/institution-location.aest.controller.ts:20]: just curious why `institution` is removed?

Comment (guru-aot) [reply sources/packages/api/src/route-controllers/institution-locations/institution-location.aest.controller.ts:20]: To have the routes, something as below when we append the Client tags

api/aest/institution/:institutionId - One institution
api/aest/institution/:institutionId/location - All locations
api/aest/location/:locationId - One location

api/institutions/institution/:institutionId - One institution
api/institutions/institution/:institutionId/location - All locations
api/institutions/location/:locationId - One location

**Review by henryoforeh-aot — APPROVED** [review]

File: sources/packages/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts [hunk 3]
```diff
@@ -63,11 +61,10 @@ import { transformAddressDetailsForAddressBlockForm } from "../utils/address-uti
  * Institution location controller for institutions Client.
  */
 @AllowAuthorizedParty(AuthorizedParties.institution)
-@Controller("institution/location")
+@Controller("location")
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts:64]: If this was changed line 65 should also be changed, right?

File: sources/packages/forms/src/form-definitions/institutionlocation.json [hunk 4]
```diff
@@ -117,7 +117,7 @@
                     "name": "Disable inputs for Client not AEST",
                     "trigger": {
                         "type": "javascript",
-                        "javascript": "result = (data.clientType !== 'aest')"
+                        "javascript": "result = (data.clientType && data.clientType !== 'aest')"
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/institutionlocation.json:120]: Why this change was introduced?

**Review by ann-aot — APPROVED** [review]
👍

File: sources/packages/forms/src/form-definitions/institutionlocation.json (hunk 4 above)
Comment (guru-aot) [reply sources/packages/forms/src/form-definitions/institutionlocation.json:120]: To enable fields when the client Type was not present, this is used for Add location

Comment (dheepak-aot) [reply sources/packages/forms/src/form-definitions/institutionlocation.json:120]: This expression can be simplified as 
` result =(data?.clientType !== 'aest'); `

**Review by dheepak-aot — APPROVED** [review]
LGTM @guru-aot. Nice work organizing the location APIs 👍
